implementation/src/main/java/io/smallrye/health/checks/NonHeapMemoryHealthCheck.java

```diff
@@ -0,0 +1,54 @@
+package io.smallrye.health.checks;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+
+/**
+ * Health check implementation that is checking memory usage against available memory
+ *
+ * <pre>
+ * {@code
+ * &#64;Produces
+ * &#64;ApplicationScoped
+ * &#64;Liveness
+ * HealthCheck check1() {
+ *   return new NonHeapMemoryHealthCheck();
+ * }
+ * }
+ * </pre>
+ */
+public class NonHeapMemoryHealthCheck implements HealthCheck {
+
+    double maxPercentage = 0.9;
+
+    public NonHeapMemoryHealthCheck() {
+    }
+
+    public NonHeapMemoryHealthCheck(double maxPercentage) {
+        this.maxPercentage = maxPercentage;
+    }
+
+    @Override
+    public HealthCheckResponse call() {
+        MemoryMXBean memoryBean = ManagementFactory.getMemoryMXBean();
+        long memUsed = memoryBean.getNonHeapMemoryUsage().getUsed();
+        long memMax = memoryBean.getNonHeapMemoryUsage().getMax();
+
+        HealthCheckResponseBuilder responseBuilder = HealthCheckResponse.named("non-heap-memory")
+                .withData("used", memUsed)
+                .withData("max", memMax)
+                .withData("max %", String.valueOf(maxPercentage));
+
+        if (memMax > 0) {
+            boolean status = (memUsed < memMax * maxPercentage);
+            return responseBuilder.state(status).build();
+        } else {
+            // Max not available
+            return responseBuilder.up().build();
+        }
+    }
+}
```

Comment:
There's a lot of duplication in this class and HeapMemoryHealthCheck. Maybe we can take out the duplication into an abstract parent class? I won't block the PR because of this. Can be done later.

Reply:
Agree. Let's do this in another PR